<launch>

  <arg name="map_frame_id" default="map"/>
  <arg name="map_size_x" default="10"/>
  <arg name="map_size_y" default="10"/>
  <arg name="map_size_z" default="5"/>


  <arg name="vicon_topic" default="/vicon/all_obs_odom"/>

  <arg name="semantic_path"  default="$(find map_generator)/semantics/3dcase1.csv"  />

  

  <node pkg ="map_generator" name ="vicon_map3d" type ="vicon_map3d" output = "screen">    

    <param name="map/x_size"     value="$(arg map_size_x)" />
    <param name="map/y_size"     value="$(arg map_size_y)" />
    <param name="map/z_size"     value="$(arg map_size_z)" />
    <param name="map/resolution" value="0.1"/>        
    <param name="map/frame_id"     value="$(arg map_frame_id)" />   

    <param name="sensing/radius" value="5.0"/>
    <param name="sensing/rate"   value="20.0"/>

    <param name="cylinder_radius"   value="0.6"/>

    
    <remap from="~vicon_all_obs" to="$(arg vicon_topic)"/>
    
    <param name="semantic_path" value="$(arg semantic_path)"/>


  </node>

</launch>
